metabase/metabase
# Support specific date filters with 00:00 time

Fixes https://github.com/metabase/metabase/issues/43057

Previously we didn't differentiate between date and date + 00:00 time filters on the FE, so 00:00 got converted to the whole date. This leads to different query results. This PR fixes the issue and moves `hasTime` from UI state to `filterParts` and computes it based on the serialized date format. This matches how QP and MBQL lib work.

How to verify:
- Open Orders
- Add filter for Created At - `On`, `November 18, 2024`
- Try it with and without 00:00 time
- Your choice should be preserved and you should get different query results.

<img width="432" alt="Screenshot 2024-06-28 at 10 49 37" src="https://github.com/metabase/metabase/assets/8542534/63dd56f4-7941-4fab-a18e-72064cafeee9">
